bids_to_msd_dataset_conversion.py

```diff
@@ -0,0 +1,290 @@
+
+Some possible options of how the datalists can be created:
+1. Single Session Single Contrast: Picks a single contrast and its corresponding GT label for each subject
+2. Single Session Multi-Contrast: Within a single session, all contrasts or list of specified contrasts are picked 
+    for each subject. Two options are provided for dealing with multiple contrasts:
+    (i) Treat the contrasts independently to create separate (image, label) pairs for each contrast, OR,
+    (ii) Group by contrasts and create a single (image, label) entity for each subject.
+    NOTE: Currently assumes that all contrasts are co-registered hence only one GT label for all 
+    the contrasts is used.
+3. Multi-Session Single Contrast: From all sessions or a list of specified sessions, the corresponding (image, label) 
+    pairs for a "single" contrast are picked. Two options are provided: 
+    (i) Treat the sessions independently to create separate (image, label) pairs for each session, OR,
+    (ii) Group by sessions and create a single (image, label) entity for each subject.
+4. Multi-Session Multi-Contrast: This not currently supported and will soon be a feature
+
+Some usage examples:
+1. Single Session Single Contrast:
+    python bids_to_msd_dataset_conversion.py --path_data /path/to/bids/dataset
+      --path_out /path/to/output/directory --split 0.6 0.2 0.2 --label_suffix _lesion-manual --include_contrasts T2w
+
+2. Single Session Multi-Contrast:
+    Assuming 5 contrasts are available (T1w, T2w, FLAIR, PD, T2star) only choose FLAIR and T2w:
+    python bids_to_msd_dataset_conversion.py --path_data /path/to/bids/dataset
+        --path_out /path/to/output/directory --split 0.6 0.2 0.2 --include_contrasts FLAIR T2w --group_by_contrasts 
+        --label_suffix _seg-lesion0 --common_label_contrast FLAIR
+
+3. Multi-Session Single Contrast:
+    Assuming 3 sessions are available (ses-01, ses-02, ses-03) only choose ses-01 and ses-03 and with sagittal images
+    python bids_to_msd_dataset_conversion.py --path_data /path/to/bids/dataset
+        --path_out /path/to/output/directory --split 0.6 0.2 0.2 --include_sessions ses-01 ses-03 --include_contrasts acq-sag_T2w
+        --group_by_sessions
+"""
+
+# TODO: 
+# - Add support for multi-session multi-contrast
+# - Check if the (image, label) pairs exist before adding them to the datalist
+
+import os
+import json
+import pandas as pd
+from tqdm import tqdm
+import numpy as np
+import argparse
+import nibabel as nib
+from loguru import logger
+from sklearn.model_selection import train_test_split
+import utils_dataset_conversion as utils
+import glob
+
+parser = argparse.ArgumentParser(description='Code for creating datalists according to the MSD format.')
+
+parser.add_argument('--seed', default=42, type=int, help="Seed for reproducibility")
+parser.add_argument('--path_data', default='./', required=True, type=str, 
```

Comment:
I think that usually `-` is commonly used over `_` in input args names, e.g., `--path-data`

Reply:
If we use `--path-data` then we can't use the same variable in the script elsewhere right? Like, with `--path_data`, I can still use `args.path_data` but if I change it to `--path-data` then does `args.path_data` still work?

Reply:
Just checked that it works! 🤯 I really didn't know this!

Reply:
Thanks for this suggestion! Changed in https://github.com/ivadomed/templates/pull/1/commits/bf260b414fbd197f56b793c394cf5b5b7108a65a!